Src/PerformanceCollector/Shared/Implementation/WebAppPerformanceCollector/WebAppPerformanceCollector.cs

```diff
@@ -0,0 +1,234 @@
+        /// <param name="error">Captures the error logged.</param>
+        /// <param name="blockCounterWithInstancePlaceHolder">Boolean that controls the registry of the counter based on the availability of instance place holder.</param>
+        public void RegisterCounter(
+            string perfCounter,
+            string reportAs,
+            bool isCustomCounter,
+            out string error,
+            bool blockCounterWithInstancePlaceHolder)
+        {
+            error = null;
+
+            try
+            {
+                bool useInstancePlaceHolder = false;
+                string parsingError = null;
+                var pc = PerformanceCounterUtility.CreateAndValidateCounter(perfCounter, null, null, out useInstancePlaceHolder, out parsingError);
+
+                if (!string.IsNullOrEmpty(parsingError))
+                {
+                    error = parsingError;
+                }
+
+                if (pc != null)
+                {
+                    this.RegisterPerformanceCounter(perfCounter, reportAs, pc.CategoryName, pc.CounterName, pc.InstanceName, useInstancePlaceHolder, false);
+                }
+                else
+                {
+                    this.RegisterPerformanceCounter(perfCounter, reportAs, string.Empty, perfCounter, string.Empty, useInstancePlaceHolder, false);
+                }
+            }
+            catch (Exception e)
+            {
+                throw new InvalidOperationException(
+                    string.Format(
+                        CultureInfo.CurrentCulture,
+                        Resources.WebAppPerformanceCounterRegistrationFailed,
+                        perfCounter),
+                    e);
+            }
+        }
+
+        /// <summary>
+        /// Rebinds performance counters to Windows resources.
+        /// </summary>
+        public void RefreshPerformanceCounter(PerformanceCounterData pcd)
+        {
+            Tuple<PerformanceCounterData, ICounterValue> tupleToRemove = this.performanceCounters.FirstOrDefault(t => t.Item1 == pcd);
+            if (tupleToRemove != null)
+            {
+                this.performanceCounters.Remove(tupleToRemove);
+            }
+
+            this.RegisterPerformanceCounter(
+                pcd.OriginalString,
+                pcd.ReportAs,
+                pcd.CategoryName,
+                pcd.CounterName,
+                pcd.InstanceName,
+                pcd.UsesInstanceNamePlaceholder,
+                pcd.IsCustomCounter);
+        }
+
+        /// <summary>
+        /// Collects a value for a single counter.
+        /// </summary>
+        private static float CollectCounter(string coutnerOriginalString, ICounterValue counter)
+        {
+            try
+            {
+                return counter.GetValueAndReset();
+            }
+            catch (Exception e)
+            {
+                 throw new InvalidOperationException(
```

Comment:
This is to explicitly throw InvalidOperationException with the specific message.